### PR TITLE
[dagster-managed-elements] Getting started guide

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Implements a first pass at a managed elements/config guide for `dagster-airbyte`, walking through implementing a basic Airbyte connection.

## Test Plan
[vercel](https://dagster-j2maijmdu-elementl.vercel.app/guides/dagster/airbyte-managed-elements)

### CODE REVIEW

**Comment by benpankow** [comment]
Current dependencies on/for this PR:
* master
  * **PR #10224** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10224" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/10224?utm_source=stack-comment).

**Comment by sryza** [comment]
Thoughts on making this part of the Airbyte integration docs or including Airbyte in the title? I think it comes off a little weird to present it as general functionality when it only works with a single integration.

**Comment by erinkcochran87** [comment]
> Thoughts on making this part of the Airbyte integration docs or including Airbyte in the title? I think it comes off a little weird to present it as general functionality when it only works with a single integration.

Will Managed elements ever be made available to other integrations? If so, I think it's fine to keep the title and location as-is. I do recommend linking to this from some place in the Airbyte docs, though.

File: docs/content/guides/dagster/airbyte-managed-elements.mdx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+## 5. Applying changes
+
+These changes look like what we expect, so we'll apply them:
+
+```bash
+dagster-me apply --module my_python_module.my_submodule:reconciler
+```
+
+Now, we should see our new connection in the Airbyte UI.
```
Comment (erinkcochran87) [inline docs/content/guides/dagster/airbyte-managed-elements.mdx:123]: It'd be cool to include a screenshot of the connection in Airbyte here.

File: docs/content/guides/dagster/airbyte-managed-elements.mdx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+```python startafter=start_new_reconciler endbefore=end_new_reconciler file=/guides/dagster/managed_elements/airbyte.py dedent=4
+airbyte_reconciler = AirbyteManagedElementReconciler(
+    airbyte=airbyte_instance,
+    connections=[cereals_connection],
+)
+```
+
+## 4. Validating changes
+
+Next, we'll use the `dagster-me` CLI to sanity check our reconciler and apply any changes. The `check` command will print out any differences between the current state of the Airbyte instance and the desired state specified in the reconciler. The CLI can be invoked by pointing it at a module which contains the reconciler.
```
Comment (erinkcochran87) [inline docs/content/guides/dagster/airbyte-managed-elements.mdx:92]:
```suggestion
Next, we'll use the `dagster-me` CLI to sanity-check our reconciler and apply any changes. The `check` command prints out differences between the current state of the Airbyte instance and the desired state specified in the reconciler. The CLI can be invoked by pointing it at a module that contains the reconciler.
```

A few small grammar tweaks here. Also suggesting active voice to improve clarity / readability.

File: docs/content/guides/dagster/airbyte-managed-elements.mdx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+```bash
+dagster-me apply --module my_python_module.my_submodule:reconciler
+```
+
+Now, we should see our new connection in the Airbyte UI.
+
+## 6. Loading connections into Dagster
+
+In order to load managed connections into Dagster, you can use the `load_assets_from_connections` utility method. This functions similarly to [`load_assets_from_airbyte_instance`](/integrations/airbyte#loading-airbyte-asset-definitions-from-an-airbyte-instance), but will validate that the connections you pass in match the connections defined in your Airbyte instance.
```
Comment (erinkcochran87) [inline docs/content/guides/dagster/airbyte-managed-elements.mdx:127]:
```suggestion
In order to load managed connections into Dagster, you can use the `load_assets_from_connections` utility method. This functions similarly to [`load_assets_from_airbyte_instance`](/integrations/airbyte#loading-airbyte-asset-definitions-from-an-airbyte-instance), but validates that the connections you pass in match the connections defined in your Airbyte instance.
```

Suggesting active voice.

File: docs/content/guides/dagster/airbyte-managed-elements.mdx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+In order to use this feature, you must first install `dagster-airbyte` and the `dagster-managed-elements` package, which includes the base config reconcilation APIs and a CLI, available as `dagster-me`.
+
+```bash
+pip install dagster-airbyte dagster-managed-elements
+```
+
+## 1. Defining a reconciler
+
+The config for your Airbyte instance is specified in a `AirbyteManagedElementReconciler`, which is pointed at a specific Airbyte instance using an Airbyte resource. It is provided with a list of connections to reconcile, which we will set up in the following steps.
```
Comment (erinkcochran87) [inline docs/content/guides/dagster/airbyte-managed-elements.mdx:21]: Thoughts on linking to the `Additional configuration options` section here?

**Review by yuhan — DISMISSED** [review]
Q management

since the DD agenda announcement is out, should we officially rename Managed Elements to Data Ingestion as Code? also, thoughts on making the guide a subpage under airbyte? so the left nav would be something like:
image

File: docs/content/guides/dagster/airbyte-managed-elements.mdx (hunk 4 above)
[the remark this answers is not in the document]
Comment (benpankow) [reply docs/content/guides/dagster/airbyte-managed-elements.mdx:21]: Provided a link out to the Airbyte guide which covers setting up the Resource in more detail.

**Review by erinkcochran87 — APPROVED** [review]
:shipit:

**Pull request merged**